tests/test_schema.py

```diff
@@ -291,6 +296,82 @@ def testJoin2(self):
         self.assertEqual(s1.join("a", "b", "c"), "a_b_c")
         self.assertEqual(s1.join("a", "b", "c", "d"), "a_b_c_d")
 
+    def _makeMapper(self, name1="a", name2="bb", name3="ccc", name4="dddd"):
+        schema = lsst.afw.table.Schema()
+        schema.addField(name1, type=float)
+        schema.addField(name2, type=float)
+        schema.addField(name3, type=float)
+        schema.addField("asdf", type="Flag")
+        mapper = lsst.afw.table.SchemaMapper(schema)
+
+        # add a default mapping for the first field
+        mapper.addMapping(schema.find(name1).key)
+
+        # add a mapping to a new field for the second field
+        field = lsst.afw.table.Field[float](name3, "doc for thingy")
+        mapper.addMapping(schema.find(name2).key, field)
+
+        # add a totally separate field to the output
+        mapper.addOutputField(lsst.afw.table.Field[float](name4, 'docstring'))
+        return mapper
+
+    def testOperatorEquals(self):
+        mapper1 = self._makeMapper()
+        mapper2 = self._makeMapper()
+        self.assertEqual(mapper1, mapper2)
+
+    def testNotEqualInput(self):
+        """Check that differing input schema compare not equal."""
+        mapper1 = self._makeMapper(name2="somethingelse")
+        mapper2 = self._makeMapper()
+        # output schema should still be equal
+        self.assertTrue(_checkSchemaIdentical(mapper1.getOutputSchema(), mapper2.getOutputSchema()))
+        self.assertNotEqual(mapper1, mapper2)
+
+    def testNotEqualOutput(self):
+        """Check that differing output schema compare not equal."""
+        mapper1 = self._makeMapper(name4="another")
+        mapper2 = self._makeMapper()
+        # input schema should still be equal
+        self.assertTrue(_checkSchemaIdentical(mapper1.getInputSchema(), mapper2.getInputSchema()))
+        self.assertNotEqual(mapper1, mapper2)
```

Comment:
It is not clear to me what these two tests are doing. The setup part looks basically identical yet one tests the input schema and one tests the output schema and it's not clear if `_makeMapper` does something fundamentally different in the two tests. If `_makeMapper` is doing basically the same thing for both tests then I suggest combining the tests to make the code less confusing. If _makeMapper is doing something different then please document what is going on.

Reply:
I thought the test docstrings were pretty clear. These are all separate tests because I needed to keep track of which ones passed and which failed as I was developing it.

Would it help if I renamed the parameters going into `_makeMapper`?

Reply:
My question is: do the two calls to _makeMapper do something fundamentally different that is relevant to the one kind of test or the other? Or are you just making different mappers for the heck of it? _makeMapper has no docs so I haven't a clue.

If there is no practical difference between the mappers for the two tests then I personally think the tests would be easier to read combined into one. If one fails the traceback will show which bit failed.

Reply:
I think our habit of putting all the tests in the same method can result in difficult development and debugging, because you can flip-flop between fixing one test and breaking another without knowing it, because the first failure masks all the others.

Reply:
Fair enough, but once the tests are written they should pass, and clarity becomes paramount. Somebody reading the tests should understand what's going on. When I read these tests I was confused.